django/contrib/admin/static/admin/js/actions.js

```diff
@@ -113,6 +113,42 @@
             list_editable_changed = true;
         });
         $('form#changelist-form button[name="index"]').click(function(event) {
+            var actionSelected = $('select[name="action"]').val(),
+                itemSelected = $('table#result_list td.action-checkbox input:checked').length > 0,
+                messageList = $('div#container ul.messagelist'),  // admin base.html's message list
+                getMessageList = function() {
+                    return (messageList.length) ? messageList : function() {
+                        // insert in the original django admin position
+                        $('<ul class="messagelist"></ul>').insertBefore('div#content');
+                        messageList = $('div#container ul.messagelist');
+                        return messageList; // return the list (<ul><>) instance
+                    }();
+                },
+                skipChangeListFormSubmit = function(msg, type, replace) {
+                    event.preventDefault();  // skip submit
+                    if (replace) {
+                        // replace the message list with the new item error
+                        getMessageList().html('<li class="' + type + '">' + msg + '</li>');
```

Comment:
@jarshwah the goal is to limit the attack surface, no matter where from the input is coming right now -- in the future this might change. My knowledge of jquery is kinda limited; but if I have chaining and everything in my head properly, something similar to this should do it:
```
getMessageList().html($('<li></li>').attr({"class": type}).text(msg));
```